### PR TITLE
Add support for NetBSD/aarch64-be (big-endian arm64).

### PR DESCRIPTION
**Pull request by he32**

### CODE REVIEW

**Comment by b-naber** [comment]
r? rust-lang/compiler

**Comment by he32** [comment]
> These commits modify **compiler targets**.
> (See the [Target Tier Policy]
> (https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html).)

Hm...  Yes...  I see there is some documentation I need to fill
out, will take a stab drafting that and including it in an
amended pull request.

Regards,

- Håvard

**Comment by WaffleLapkin** [comment]
Marked this as waiting on author, feel free to `@rustbot ready` once you filled out everything regarding the target tier policy and fixed CI.
r? compiler

**Comment by Urgau** [comment]
As said by CI you are missing an the entry in the [platform support page](https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support.md) (`src/doc/rustc/src/platform-support.md`), with the page for the target in `src/doc/rustc/src/platform-support/`. Take a look at the [template](https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support/TEMPLATE.md) and an [example](https://github.com/rust-lang/rust/pull/108792/files#diff-98c317f447306855c6e723385f2a38b516acb20296d4131b5572a2ca1b112add) for what is roughly expected.

**Comment by he32** [comment]
I committed above a suggested update to `platform-support.md`, and added a `netbsd.md` file, partially patterned after the OpenBSD file.  Almost all of the NetBSD targets are suggested as "Tier 3" systems, and do not have CI setups configured.
Hope this is a step in the right direction.

File: src/doc/rustc/src/platform-support/netbsd.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+For the other systems mentioned above, using the `pkgsrc` route is
+probably the easiest, possibly via the `rust-bin` package to save
+time, see the `RUST_TYPE` variable from the `rust.mk` Makefile
+fragment.
+
+The pkgsrc rust package has a few files to assist with building
+pkgsrc packages written in rust, ref. the `rust.mk` and `cargo.mk`
+Makefile fragments in the `lang/rust` package.
+
```
Comment (wesleywiser) [inline src/doc/rustc/src/platform-support/netbsd.md:108]:
```suggestion
```

**Comment by he32** [comment]
> The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5346903376/jobs/9694581217) [(plain)](https://github.com/rust-lang/rust/commit/da05ed36836077392384573d5277f8a01bbe17db/checks/14474512455/logs)
> Click to see the possible cause of the failure (guessed by this bot)

I'll admit that I don't quite understand the criterion for the "broken link" messages.
The links I inserted in the source document are all possible to visit, so what is "broken"
about them?

**Comment by oli-obk** [comment]
I don't know if the rustc guide has any guidance for that, but I'd just look at how other links work and check if

a) the link should be relative to the file
b) the link should have a different extension

**Comment by Urgau** [comment]
@he32 I think you are missing the definition in the `SUMMARY.md` file: [`src/doc/rustc/src/SUMMARY.md`](https://github.com/rust-lang/rust/blob/da05ed36836077392384573d5277f8a01bbe17db/src/doc/rustc/src/SUMMARY.md); without it mdbook [does not know](https://rust-lang.github.io/mdBook/format/summary.html) how to include files (or anything for that matter).

**Pull request merged**